File: benchmarks/graphviz/graphviz-2.34.0/tclpkg/gv/demo/modgraph.lua
#!/usr/bin/lua

-- display the kernel module dependencies

-- author: John Ellson <ellson@research.att.com>

require('gv')

modules = io.lines('/proc/modules') 

G = gv.digraph("G")
N = gv.protonode(G)
E = gv.protoedge(G)
gv.setv(G, 'rankdir', 'LR')
gv.setv(G, 'nodesep', '0.05')
gv.setv(N, 'shape', 'box')
gv.setv(N, 'width', '0')
gv.setv(N, 'height', '0')
gv.setv(N, 'margin', '.03')
gv.setv(N, 'fontsize', '8')
gv.setv(N, 'fontname', 'helvetica')
gv.setv(E, 'arrowsize', '.4')

for rec in modules do
   for mod, usedbylist in string.gfind(rec, "([_%w]+) %w+ %w+ ([-,_%w]+)") do
      n = gv.node(G, mod)
      for usedby in string.gfind(usedbylist, "([-_%w]+)") do
         if (usedby ~= '-') and (usedby ~= '') then
            gv.edge(n, gv.node(G, usedby))
         end
      end
   end
end

gv.layout(G, 'dot')
gv.render(G, 'gtk')

File: benchmarks/graphviz/graphviz-2.34.0/tclpkg/gv/demo/modgraph.php
#!/usr/bin/php
<?php
include("gv.php");

# display the kernel module dependencies

# author: John Ellson <ellson@research.att.com>

$G = gv::digraph("G");
$N = gv::protonode($G);
$E = gv::protoedge($G);

gv::setv($G, "rankdir", "LR");
gv::setv($G, "nodesep", "0.05");
gv::setv($N, "shape", "box");
gv::setv($N, "width", "0");
gv::setv($N, "height", "0");
gv::setv($N, "margin", ".03");
gv::setv($N, "fontsize", "8");
gv::setv($N, "fontname", "helvetica");
gv::setv($E, "arrowsize", ".4");

if ($f = fopen("/proc/modules", "r")) while ($rec = fgets($f)) {
	$matches = preg_split("/[\s]+/", $rec, -1, PREG_SPLIT_NO_EMPTY);
	$n = gv::node($G,$matches[0]);
	$usedbylist = preg_split("/[,]/", $matches[3], -1, PREG_SPLIT_NO_EMPTY);
	foreach ($usedbylist as $i => $usedby) {
		if ($usedby != "-") {
			gv::edge($n, gv::node($G, $usedby));
		}
	}
}
fclose($f);

gv::layout($G, "dot");
gv::render($G, "xlib");

?>
